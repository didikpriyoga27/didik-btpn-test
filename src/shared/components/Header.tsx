import {NavigationProp, useNavigation} from '@react-navigation/native';
import {useColorScheme} from 'nativewind';
import React, {FC} from 'react';
import {useSafeAreaInsets} from 'react-native-safe-area-context';

import {StackParamList} from '../../navigation/StackParamList';
import BackButtonIcon from '../assets/svg/BackButtonIcon';
import Text from './Text';
import TouchableOpacity from './TouchableOpacity';
import View from './View';

type Props = {
  title?: string;
  isWithBackButton?: boolean;
};
const Header: FC<Props> = ({title, isWithBackButton = false}) => {
  const {goBack} = useNavigation<NavigationProp<StackParamList>>();
  const {top} = useSafeAreaInsets();
  const {colorScheme} = useColorScheme();

  const isDark = colorScheme === 'dark';

  if (!isWithBackButton && !title) {
    return null;
  }

  return (
    <View
      className={'z-10 w-full justify-end'}
      // eslint-disable-next-line react-native/no-inline-styles
      style={{
        marginTop: -top,
        height: title ? top + 48 : top + 80,
        elevation: 4,
      }}>
      <View className={`flex-row items-center px-6 ${title ? 'h-12' : 'h-20'}`}>
        {title && (
          <Text className="absolute left-0 right-0 flex-1 text-center text-base">
            {title}
          </Text>
        )}
        {isWithBackButton && (
          <TouchableOpacity onPress={goBack} className="-ml-2 p-2">
            <BackButtonIcon color={isDark ? 'white' : 'black'} />
          </TouchableOpacity>
        )}
      </View>
    </View>
  );
};

export default Header;
