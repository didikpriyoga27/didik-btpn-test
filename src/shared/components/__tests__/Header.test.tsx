import {render} from '@testing-library/react-native';
import React from 'react';

import Header from '../Header';

jest.useRealTimers();

describe('@shared/components/Header', () => {
  describe('when header is null', () => {
    const component = () => render(<Header />);

    test('snapshot: component', async () => {
      const {toJSON} = component();
      expect(toJSON()).toMatchSnapshot();
    });
  });
  describe('when header is rendered but no title', () => {
    const component = () => render(<Header isWithBackButton />);

    test('snapshot: component', async () => {
      const {toJSON} = component();
      expect(toJSON()).toMatchSnapshot();
    });
  });
});
