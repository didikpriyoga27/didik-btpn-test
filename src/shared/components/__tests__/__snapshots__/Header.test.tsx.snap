// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@shared/components/Header when header is null snapshot: component 1`] = `null`;

exports[`@shared/components/Header when header is rendered but no title snapshot: component 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#fff",
      },
      [
        {
          "zIndex": 10,
        },
        {
          "width": "100%",
        },
        {
          "justifyContent": "flex-end",
        },
        {
          "elevation": 4,
          "height": 80,
          "marginTop": -0,
        },
      ],
    ]
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "#fff",
        },
        [
          {
            "flexDirection": "row",
          },
          {
            "alignItems": "center",
          },
          {
            "paddingLeft": 24,
            "paddingRight": 24,
          },
          {
            "height": 80,
          },
        ],
      ]
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#fff",
          "marginLeft": -8,
          "opacity": 1,
          "paddingBottom": 8,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 8,
        }
      }
    >
      <RNSVGSvgView
        bbHeight={24}
        bbWidth={24}
        color="black"
        fill="none"
        focusable={false}
        height={24}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "flex": 0,
              "height": 24,
              "width": 24,
            },
          ]
        }
        tintColor="black"
        width={24}
      >
        <RNSVGGroup
          fill={null}
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="M2 12h20M2 12l6-6M2 12l6 6"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            propList={
              [
                "stroke",
                "strokeWidth",
                "strokeLinecap",
              ]
            }
            stroke={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            strokeLinecap={1}
            strokeWidth={2}
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
  </View>
</View>
`;
